### PR TITLE
Add markdown language definition

### PR DESCRIPTION
**Pull request by anctartica**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by MacBox7** [comment]
Why 2 commits? Squash them into one.

**Comment by anctartica** [comment]
I created 2 files here. Which filename would you use in the commit message?

**Comment by MacBox7** [comment]
IMHO single commit with shortlog => `Definations: Add new language definations`
Would work quite well.

File: coalib/bearlib/languages/definitions/flow.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class flow:
```
Comment (li-boxuan) [inline coalib/bearlib/languages/definitions/flow.py]: I would suggest capitalizing the initial letter of the class name. (ref: https://www.python.org/dev/peps/pep-0008/#class-names)

File: coalib/bearlib/languages/definitions/markdown.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class markdown:
```
Comment (li-boxuan) [inline coalib/bearlib/languages/definitions/markdown.py]: Same here. Change markdown into Markdown. Btw, Markdown is always written with an upper case M. That means you should also change your filename.

File: coalib/bearlib/languages/definitions/Markdown.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Markdown:
+    extensions = [
+        '.markdown', '.mdown', '.mkdn',
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/Markdown.py]: not `.text` - that will cause too many files to be linted using markdown, which will cause all sorts of errors on normal text files.

Also I wouldnt want to assume that `.Rmd`'s syntax for the code blocks is legal markdown (based on which spec of markdown ... ? ;-)  IMO, exclude it, and someone else can worry about that problem.

File: coalib/bearlib/languages/definitions/Flow.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Flow:
+    extensions = '.js'
+    comment_delimiter = '//'
+    multiline_comment_delimiters = {'/*': '*/'}
+    string_delimiters = {'"': '"', "'": "'"}
+    multiline_string_delimiters = {'`': '`'}
```
[the remark this answers is not in the document]
Comment (anctartica) [reply coalib/bearlib/languages/definitions/Flow.py]: created #5343

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5279/commits/0fe3194a5d7ef882f87c8e22919f72a72a2fceb4

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**